### PR TITLE
Add static list of reserved names

### PR DESCRIPTION
**Pull request by Elhebert**
### What does it do?

Add a list of reserved names.

I've tested the values we had in v3 and removed the ones that were no longer breaking Strapi. I might have missed edge cases thought. I tried some MySQL reserved words as well to see if we needed to add them to the list, but we are good on that side.

### CODE REVIEW

File: packages/core/content-type-builder/server/services/builder.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
 'use strict';
 
 module.exports = () => ({
-  // TODO: Implement
   getReservedNames() {
     return {
-      models: [],
-      attributes: [],
+      models: ['boolean', 'date', 'date-time', 'time', 'upload'],
+      attributes: ['id', 'created_at', 'updated_at', 'created_by', 'updated_by', 'published_at'],
```
Comment (alexandrebodin) [inline packages/core/content-type-builder/server/services/builder.js:7]: You will need the camelCase values too

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

**Review by alexandrebodin — APPROVED** [review]
Good 1st version to start with

**Comment by derrickmehaffy** [comment]
@Elhebert is this based on this tracking issue?
https://github.com/strapi/strapi/issues/1198

**Comment by Elhebert** [comment]
@derrickmehaffy I was not aware of this issue. 
@alexandrebodin should I go through all the strings in the issue to see if we still need to safeguard them?

**Comment by alexandrebodin** [comment]
I don't think you need to as you said we won't have problem with pluralization of the CTs. 

Let's see if the pbl is coming from the main branch maybe. I'm pinging the FE team

**Comment by Elhebert** [comment]
In case it's needed here's the same unit test locally with node 14:
Screen Shot 2021-11-03 at 10 20 41 AM

**Comment by alexandrebodin** [comment]
let's merge and check on the main PR

**Pull request merged**